src/util/strb.h

```diff
@@ -159,7 +159,21 @@ static inline void strb_appendb(strb *sb, strb *sb2) {
  *
  * A format error will place the strb in error mode.
  */
-GPUARRAY_LOCAL void strb_appendf(strb *, const char *f, ...);
+void strb_appendf(strb *, const char *f, ...);
+
+/*
+ * Reads from the file specified by the given file descriptor.
+ *
+ * A read error will place the strb in error mode.
+ */
+void strb_read(strb *, int fd, size_t sz);
```

Comment:
We don't need to specify parameter names on declaration, so this wouldn't be a problem.  However it jars with the style of the rest of the file so I'll fix it.